src/kixi/datastore/schemastore/elasticsearch.clj

```diff
@@ -104,10 +106,16 @@
     component/Lifecycle
     (start [component]
       (if-not conn
-        (let [[host port] (if discover (es/discover-executor discover) [host port])
-              connection (es/connect host port)
+        (let [{:keys [native-host-ports http-host-ports]} (if discover
```

Comment:
it feels like the connection bit and  in metadatastore and schemastore elasticsearch components are pretty much the same (except for the database name). Would it make sense to refactor this out to avoid duplication?